### PR TITLE
Apply projectHealth recommendations to many remaining JVM and software projects

### PR DESCRIPTION
**Pull request by tresat**
This addresses the following projects:

  - scala
  - toolchains-jvm
  - publish
  - resources-gcs
  - resources-http
  - resources-s3
  - resources-sftp
  - security
  - signing
  - version-control

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
This test failure on CI can be ignored - this is ready for review

File: platforms/software/resources-gcs/build.gradle.kts [hunk 1]
```diff
@@ -12,25 +12,33 @@ errorprone {
...
+    implementation(project(":logging-api"))
 
-    implementation(libs.slf4jApi)
-    implementation(libs.guava)
     implementation(libs.commonsLang)
-    implementation(libs.gcs)
+    implementation(libs.googleApiClient)
+    implementation(libs.googleHttpClientGson)
+    implementation(libs.googleHttpClient)
+    implementation(libs.googleOauthClient)
```
[the remark this answers is not in the document]
Comment (tresat) [reply platforms/software/resources-gcs/build.gradle.kts:28]: Same situation as above.

File: platforms/software/resources-s3/build.gradle.kts [hunk 2]
```diff
@@ -15,25 +15,26 @@ errorprone {
...
-    implementation(libs.awsS3Core)
-    implementation(libs.awsS3S3)
-    implementation(libs.awsS3Kms) {
+    api(libs.awsS3Core)
+    api(libs.awsS3S3)
+    api(libs.awsS3Kms) {
         because("Loaded by the AWS libraries with reflection when present")
     }
-    implementation(libs.awsS3Sts) {
+    api(libs.awsS3Sts) {
```
[the remark this answers is not in the document]
Comment (tresat) [reply platforms/software/resources-s3/build.gradle.kts:29]: I think we do this to ensure it is included in the distribution.

File: platforms/software/security/build.gradle.kts [hunk 3]
```diff
@@ -13,19 +13,18 @@ errorprone {
...
+
+    implementation(libs.bouncycastleProvider)
     implementation(libs.guava)
     implementation(libs.inject)
-
-    api(libs.bouncycastlePgp)
-
-    implementation(libs.groovy) {
-        because("Project.exec() depends on Groovy")
-    }
```
[the remark this answers is not in the document]
Comment (tresat) [reply platforms/software/security/build.gradle.kts:28]: We're not using any Groovy types directly in the security project and something else is responsible for adding it to the distribution.

**Comment by tresat** [comment]
@bot-gradle test this

**Review by octylFractal — APPROVED** [review]

**Comment by tresat** [comment]
@bot-gradle test and merge

**Pull request merged**